bokehjs/src/lib/models/tools/inspectors/hover_tool.ts

```diff
@@ -445,11 +445,38 @@ export class HoverToolView extends InspectToolView {
     const swatch_els = el.querySelectorAll<HTMLElement>("[data-swatch]")
 
     const color_re = /\$color(\[.*\])?:(\w*)/
+    const swatch_re = /\$swatch:(\w*)/
 
     for (const [[, value], j] of enumerate(tooltips)) {
-      const result = value.match(color_re)
-      if (result != null) {
-        const [, opts="", colname] = result
+      let tooltipValue = value
+
+      const colorFieldMatch = value.match(color_re)
+      const swatchFieldMatch = value.match(swatch_re)
+
+      if (swatchFieldMatch) {
+        tooltipValue = value.replace(swatch_re, "")
+        const [, colname] = swatchFieldMatch
+        const column = ds.get_column(colname)
+
+        display(swatch_els[j])
+
+        if (column == null) {
+          swatch_els[j].textContent = `(unknown)`
+          textOnly(swatch_els[j])
+        } else {
+          const color = isNumber(i) ? column[i] : null
+
+          if (color == null) {
+            swatch_els[j].textContent = "(null)"
+            textOnly(swatch_els[j])
```

Comment:
I think for swatches we should just silently display nothing if the value is null. Otherwise if both `$color` and `$swatch` are used it will show up as `(null) (null)` which is not great.

Reply:
sure, updated